recipes-bsp/uefi/tegra-uefi-capsules_35.5.0.bb

```diff
@@ -2,54 +2,67 @@ DESCRIPTION = "Generate UEFI capsules for bup paylods"
 LICENSE = "MIT"
 LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
 
-inherit tegra-bup deploy
+inherit tegra-bup deploy image_types_tegra kernel-artifact-names
 
 TEGRA_UEFI_CAPSULE_SIGNING_CLASS ??= "tegra-uefi-capsule-signing"
 inherit ${TEGRA_UEFI_CAPSULE_SIGNING_CLASS}
 
 TEGRA_UEFI_CAPSULE_SIGNING_EXTRA_DEPS ??= ""
 
 COMPATIBLE_MACHINE = "(tegra)"
-
-DEPENDS += "tegra-bup-payload"
+TEGRA_SIGNING_EXTRA_DEPS ??= ""
 
 GUID:tegra194 ?= "be3f5d68-7654-4ed2-838c-2a2faf901a78"
 GUID:tegra234 ?= "bf0d4599-20d4-414e-b2c5-3595b1cda402"
 
 do_compile() {
-    sign_uefi_capsules
+
+    if [ ! -z "${INITRAMFS_IMAGE}" ]; then
```

Comment:
Everything below here should be run regardless of whether INITRAMFS_IMAGE is defined or not.  Instead, the parameter passed to `oe_make_bup_payload` will differ, I think, depending on whether we have a bare kernel/kernel with bundled initramfs or a kernel with separate initrd.  Either case would still be a cboot file, but the base file name would be different.